File: myutils/src/main/java/com/jty/myutils/utils/APPUtils.java
package com.jty.myutils.utils;

import android.content.Context;
import android.content.pm.PackageInfo;

/** 
 * APP辅助类
 * @author jty 
 */

public class APPUtils {

	/**
	 * 获取应用程序名称
	 * @param context
	 * @return
     */
	public static String getAppName(Context context){
		try{
			int labelRes = context.getPackageManager()
					.getPackageInfo(context.getPackageName(), 0)
					.applicationInfo
					.labelRes;

			return context.getResources().getString(labelRes); 
		}catch(Exception e){
			e.printStackTrace();
		}
		return null;
	}
	
	
	/**
	 * 获取版本号
	 * @param context
	 * @return
	 */
	public static String getVersionName(Context context){
		try{
			PackageInfo packageInfo=context.getPackageManager()
					.getPackageInfo(context.getPackageName(), 0);
			
			return packageInfo.versionName;
			
		}catch(Exception e){
			e.printStackTrace();
		}
		return null;
	}
	/**
	 * 获得版本码
	 * @param context
	 * @return
	 */
	public static int getVersionCode(Context context){
		try{
			return context.getPackageManager()
					.getPackageInfo(context.getPackageName(), 0)
					.versionCode;
		}catch(Exception e){
			e.printStackTrace();
		}
		return 0 ;
	}

}

File: mykotlinutils/src/main/java/com/jty/mykotlinutils/utils/APPUtils.kt
package com.jty.mykotlinutils.utils

import android.content.Context
import android.content.pm.PackageInfo

/**
 * Created by TaoYuan on 2017/9/20 0020.
 *
 * Get APP Basic Args
 */

object APPUtils {

    /**
     * Get APP Name
     * @param context
     * @return string
     */
    fun getAppName(context: Context): String? {
        try {
            val labelRes = context.packageManager
                    .getPackageInfo(context.packageName, 0)
                    .applicationInfo
                    .labelRes

            return context.resources.getString(labelRes)
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return null
    }


    /**
     * Get APP Version Name
     * @param context
     * @return string
     */
    fun getVersionName(context: Context): String? {
        try {
            val packageInfo = context.packageManager
                    .getPackageInfo(context.packageName, 0)

            return packageInfo.versionName

        } catch (e: Exception) {
            e.printStackTrace()
        }

        return null
    }

    /**
     * Get APP Version Code
     * @param context
     * @return int
     */
    fun getVersionCode(context: Context): Int {
        try {
            return context.packageManager
                    .getPackageInfo(context.packageName, 0)
                    .versionCode
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return 0
    }
}
